drivers/gpio/gpio_nrf5.c

```diff
@@ -233,6 +228,21 @@ static int gpio_nrf5_config(struct device *dev,
 		u32_t config = 0;
 		u32_t port = GPIO_PORT(dev);
 
+		/* check if already allocated to replace */
+		int i = gpiote_find_channel(dev, pin, port);
+
+		if (i < 0) {
+			if (popcount(gpiote_chan_mask) == GPIOTE_CH_NUM) {
+				return -EIO;
+			}
+
+			/* allocate a GPIOTE channel */
+			i = find_lsb_set(~gpiote_chan_mask) - 1;
+		}
+
+		gpiote_chan_mask |= BIT(i);
```

Comment:
This should be inside the `if` above, since it is only needed for newly allocated channels.

Reply:
agreed